xrdp/xrdp_login_wnd.c

```diff
@@ -742,6 +745,17 @@ xrdp_login_wnd_get_monitor_dpi(struct xrdp_wm *self)
     return result;
 }
 
+void
+split_string_by_newline(const char *str, struct list *lines)
+{
+    char *copy = strdup(str);
+    char *pch = strtok(copy, "\n");
+    while (pch != NULL)
+    {
+        list_add_item(lines, (tintptr) pch);
+        pch = strtok(NULL, "\n");
+    }
+}
 
```

Comment:
Memory problems here I think. You're allocating memory with a single call to `strdup()`, and then expecting `auto_free` to fix it for you on `list_delete()`. However, you'll be calling free on addresses mid-way through the allocated block.

How about (in string_calls.c or list.c) :-

```c
static const char *
append_fragment(const char *start, const char *end, struct list *list)
{
    unsigned int len = end - start;
    char *copy = malloc(len + 1);
    g_memcpy(copy, start, len);
    copy[len] = '\0';
    list_add_item(list, (tintptr)copy);
    return end + 1;
}

struct list *
split_string_by_newline(const char *str)
{
    struct list *result = list_create();
    result->auto_free = 1;

    if (str != NULL)
    {
        const char *p;

        while ((p = g_strchr(str, '\n')) != NULL)
        {
            str = append_fragment(str, p, result);
        }

        if (*str != '\0')
        {
            append_fragment(str, str + g_strlen(str), result);
        }
    }
    
    return result;
}
```

It's got a slightly different signature admittedly, but all memory is freed on `list_delete()`

Reply:
Also, it behaves differently from `strtok()` for contibuous delimiters.

Reply:
This is what happens when you've spent too much of your career in languages with garbage collectors and then go code in an unmanaged memory language when you're tired. You're right of course!

I think I'll split this part up into smaller PRs as always to make them easier to review.